libcob/fileio.c

```diff
@@ -4150,6 +4179,23 @@ indexed_file_delete (cob_file *f, const char *filename)
 #endif
 }
 
+static int
+bdb_bt_compare(DB *db, const DBT *k1, const DBT *k2)
+{
+	int res = 0;
+	const unsigned char *col = (const unsigned char *)DBT_GET_APP_DATA(k1);
+	if (col == NULL) {
```

Comment:
Is there a reason to register `bdb_bt_compare` down there if `col == NULL`? I _think_ this check should be done outside.
Also: is it possible in our code that both DBT can have a different size? At least in the sort comparison they are always identical (and it would be a status 39 error if they wouldn't be here, no)?

Reply:
I guess if no keys for the current file have a collation, then we don't need to register `bdb_bt_compare` indeed.

Reply:
I've thought this was registered per key; in this case if the key has no collation, then we don't need to register, and can get rid of the `col == NULL` check, otherwise it needs to stay in and we only don't register if there's no collation at all.

Reply:
Oh, wait, you're right, the registration is indeed per key. So yeah, we can simplify the code further and register the comparison function only if the key has a collating sequence.

Reply:
That's part of the beauty of code-reviews - the code gets improved even before it gets live :-)
Wait, who reviews my code...

Reply:
> Also: is it possible in our code that both DBT can have a different size? At least in the sort comparison they are always identical (and it would be a status 39 error if they wouldn't be here, no)?

Yeah, they should always be of identical length in our context (and I made a quick check by adding an assert and running both NIST and the testsuite with no error). Anyway I prefered to handle the case anyways (the comparison function is rather "low-level" and does not really care about our use case).